### PR TITLE
LCAM 707 Get Contribution Summary Data

### PR DESCRIPTION
**Pull request by nigelpain**
## What

[Link to story](https://dsdmoj.atlassian.net/jira/software/projects/LCAM/boards/881?selectedIssue=LCAM-707)

Added new endpoint to retrieve data on contributions for provided repId and any associated contribution files in order to build a contribution summary object in the C3 service.

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: maat-court-data-api/src/test/java/gov/uk/courtdata/integration/contribution/ContributionsControllerIntegrationTest.java [hunk 1]
```diff
@@ -180,4 +180,19 @@ void givenAValidRepIdAndEmptyCorrespondence_whenGetContributionCountIsInvoked_th
...
                 get(ENDPOINT_URL + "/" + INVALID_REP_ID).contentType(MediaType.APPLICATION_JSON)));
     }
+
+    @Test
+    void givenValidRepId_whenGetContributionsSummaryIsInvoked_thenCorrectResponseIsReturned() throws Exception {
+        MvcResult result = runSuccessScenario(get(ENDPOINT_URL + "/" + TestModelDataBuilder.REP_ID
+                + "/summary").contentType(MediaType.APPLICATION_JSON));
+
+        Assertions.assertThat(result.getResponse().getContentAsString())
+                .contains("CONTRIBUTIONS_202307111234.xml");
```
Comment (loganathanMOJ) [inline maat-court-data-api/src/test/java/gov/uk/courtdata/integration/contribution/ContributionsControllerIntegrationTest.java:190]: Can we assert all the field with objectMapper.writeValueAsString()?

File: maat-court-data-api/src/main/java/gov/uk/courtdata/contribution/projection/ContributionsSummaryView.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    BigDecimal getMonthlyContributions();
+    BigDecimal getUpfrontContributions();
+    String getBasedOn();
+    String getUpliftApplied();
+    LocalDate getEffectiveDate();
+    LocalDate getCalcDate();
+    String getFileName();
+    LocalDate getDateSent();
+    LocalDate getDateReceived();
+}
```
Comment (loganathanMOJ) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/contribution/projection/ContributionsSummaryView.java:19]: Code smell. Please Remove unused import

File: maat-court-data-api/src/test/resources/schema-h2.sql [hunk 3]
```diff
@@ -21,7 +21,26 @@ CREATE TABLE TOGDATA.CONFIG_PARAMETERS
...
+ "USER_CREATED" VARCHAR(250),
+ "DATE_MODIFIED" DATE,
+ "XML_CONTENT" CLOB(2000),
+ "USER_MODIFIED" VARCHAR(250),
+ "DATE_SENT" DATE,
+ "DATE_RECEIVED" DATE,
+ "ACK_XML_CONTENT" CLOB(2000));
+
+INSERT INTO TOGDATA.CONTRIBUTION_FILES
+VALUES (1, 'CONTRIBUTIONS_202307111234.xml', 0, null, '2023-07-01', 'TEST', null, null, null, null, null, null);
```
[the remark this answers is not in the document]
Comment (nigelpain) [reply maat-court-data-api/src/test/resources/schema-h2.sql]: Ah yes Athar added these in didn't he but wrote this before I found out =) Good catch, I've removed the sql from the above file and added some logic into the integration tests to setup a contributions files row.

File: maat-court-data-api/src/test/java/gov/uk/courtdata/integration/contribution/ContributionsControllerIntegrationTest.java (hunk 1 above)
Comment (nigelpain) [reply maat-court-data-api/src/test/java/gov/uk/courtdata/integration/contribution/ContributionsControllerIntegrationTest.java:195]: I tried this but the assertion fails as the object built from the projection has the fields in a different order than that returned in the response. I couldn't find a way to affect the field ordering for the projection so have left it as is. Though if you have know how to solve I'm all ears =)

File: maat-court-data-api/src/main/java/gov/uk/courtdata/contribution/projection/ContributionsSummaryView.java (hunk 2 above)
Comment (nigelpain) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/contribution/projection/ContributionsSummaryView.java:18]: Removed the unused import.

**Review by loganathanMOJ — APPROVED** [review]

**Pull request merged**